### DISCUSSION TITLE
Running Code from Private Repo and Publishing Output to Pulic Repo Help

### DISCUSSION THREAD

**Discussion by albert-marrero** [community/community#25614] [Actions]
<p><strong>Background:</strong> I have a private code repo that generates a CSV and I would like to publish/push that CSV to a public code repo using Workflows/Actions on a schedule.</p>
<p><strong>Problem:</strong> Once I get the CSV generated and into the public repository and try to push. I am getting a error of unable to access “repo” denied to github-actions[bot]. Any help will be appreciated, I tried to google a solution or work around.</p>
<p>Below is the sample code:</p>
<pre><code class="lang-auto"># This is a basic workflow to help you get started with Actions

name: FooBar

# Controls when the workflow will run
on:
  # Triggers the workflow on push 
  [push]

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "foo"
  foo:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      - name: Checkout Private Repo
        uses: actions/checkout@v2
        with:
          path: main

      - name: Checkout Public Repo
        uses: actions/checkout@v2
        with:
          repository: public-repo
          path: data

      - name: Run CSV Script and Move File to Data Path
        run: |
          cd main
          python -m pip install -r requirements.txt
          python main.py

      - name: Checkin Public Repo
        run: |
          cd data
          git config user.name github-actions
          git config user.email github-actions@github.com
          git add .
          git commit -m "generated"
          git push     
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="albert-marrero" data-post="1" data-topic="201357">
<div class="title">
<div class="quote-controls"></div>
 albert-marrero:</div>
<blockquote>
<pre><code class="lang-auto">      - name: Checkout Public Repo
        uses: actions/checkout@v2
        with:
          repository: public-repo
          path: data
</code></pre>
</blockquote>
</aside>
<p>This means you’re using the <code>GITHUB_TOKEN</code> for authorization when accessing the public repository. For fetching that’s no problem (because a public repo can be fetched by anyone), but pushing won’t work because the <code>GITHUB_TOKEN</code> is scoped to your private repository.</p>
<p>To push to the public repository you’ll have to create a PAT that grants access for pushing to your public repository (<strong>repo</strong> scope), store it as a secret for your private repository, and provide that secret to <code>actions/checkout</code> as its <code>token</code> option.</p>

**Comment by albert-marrero** [comment]
<p>As much I don’t like to create a PAT that allows access to all my public repositories, this works greatly. Just to lock it down, I need to look into how to repo scope.</p>

**Comment by airtower-luna** [comment]
<p>I meant the repo scope for tokens, see: <a href="https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes" class="inline-onebox">Scopes for OAuth Apps - GitHub Docs</a></p>
<p>I don’t think there’s a way to limit it to a specific repository, unfortunately.</p>
